shelvery_tests/conftest.py

```diff
@@ -0,0 +1,96 @@
+import pytest
+import boto3
+import os
+import time
+from shelvery.aws_helper import AwsHelper
+from botocore.exceptions import ClientError
+
+from shelvery_tests.cleanup_functions import cleanupSnapshots
+
+source_account = None
+destination_account = None
+
+def pytest_addoption(parser):
+    parser.addoption("--destination", action="store", default="None")
+    parser.addoption("--source", action="store", default="None")
+
+
+def pytest_configure(config):
+    global source_account
+    global destination_account
+    
+    source_account = config.getoption('--source')
+    destination_account = config.getoption('--destination')
+
+#Add some check here on if the stack aleady exists (eg: deleting/creating/available)
+        
+@pytest.fixture(scope="session", autouse=True)
+def setup(request):
+
+    sts = AwsHelper.boto3_client('sts')
+    id = str(sts.get_caller_identity()['Account'])
+
+    if id == source_account:
+
+        cfclient = boto3.client('cloudformation')
+
+        #Check whether stack aleady exists
+        try:
+            shelvery_status = cfclient.describe_stacks(StackName='shelvery-test')['Stacks'][0]['StackStatus']
+        except ClientError as error:
+            if error.response["Error"]["Code"] == "ValidationError":
+                shelvery_status = "NONE"
+
+        # If stack aleady exists (Deleting from previous run)
+        while shelvery_status == 'DELETE_IN_PROGRESS' or shelvery_status == 'DELETE_COMPLETE':
```

Comment:
i would use the inbuilt boto3 waiter
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudformation.html#CloudFormation.Waiter.StackDeleteComplete

Reply:
Added 'Delete Waiter'